pyiron_base/generic/hdfio.py

```diff
@@ -31,6 +31,35 @@
 __status__ = "production"
 __date__ = "Sep 1, 2017"
 
+_special_symbol_replacements = {
+    '.': 'd',
+    '-': 'm',
+    '+': 'p',
+    ',': 'c',
+}
+
+
+def _get_safe_job_name(name, ndigits=8, extension=None):
+    def round_(value, ndigits=ndigits):
+        if isinstance(value, float) and ndigits is not None:
+            return round(value, ndigits=ndigits)
+        return value
+    if not isinstance(name, str):
+        name_rounded = [round_(nn) for nn in name]
+        job_name = '_'.join([str(nn) for nn in name_rounded])
+    else:
+        job_name = name
+    if extension is not None and job_name.endswith(extension):
+        job_name = job_name[:job_name.rfind(extension)]
+    job_name = job_name.split('/')
```

Comment:
No this is just a security measure in order not to change the path. So if we get something like the one above: `/cmmc/u/samsstud/dev_sam/pyiron_base/tests/generic/../static/dft`, we should make sure that it doesn't erroneously change the path. Otherwise we could simply skip the replacement altogether as soon as `/` is detected.

Reply:
Actually we shouldn't need it in the first place in `ProjectHDFio`, but I saw that the job name is directly passed there if it's created via `pr.create_job` and not via `pr.create.job`.

Reply:
Then I would be in favor of sanitizing the job_name in `create_job` and exclude any paths from conversion in this function, i.e. ValueError on `/`, `\`, `\\`.

Reply:
ok I'm gonna do it :)